chapter8.adoc

```diff
@@ -1,27 +1,66 @@
 [[chapter8]]
 [[Smsdedbg]]
-== `Smsdedbg`: Supervisor Domain External Debug
+== Supervisor Domain External Trace and Debug
 
-The <<MSDCFG>> CSR contains the `sdedbgalw` bit that controls whether the
-current scheduled SD is allowed to be external-debugged cite:[ExtDbg]. This bit
-is context switched (along with rest of the `msdcfg`) per SD.
+The <<MSDCFG>> CSR contains the `sdedbgalw` and `sdetrcalw` bits that control
+whether the current scheduled SD is allowed to be external-debugged or to be
+external-traced respectively.
 
-=== `Smsdedbg` interaction with external debug security controls (Informative)
+=== `Smsdedbg`: External Debug allowed control for Supervisor Domain
+
+When M-mode external debug is enabled, all supervisor domains may also be
+debugged by an external debugger irrespective of the configuration held in
+`msdcfg.SDEDBGALW`.
+
+When M-mode external debug is disabled, whether execution at privilege modes
+less than `M-mode` may be debugged by an external debugger depends on the
+configuration held in `msdcfg.SDEDBGALW`.
+
+When `msdcfg.SDEDBGALW` = 0, external debug is disallowed for the SD. Abstract
+commands and halt request from the debug module are suppressed and stay pending
+while the SD is active.
+
+When `msdcfg.SDEDBGALW` = 1 then external debug of privilege modes less than
+`M-mode` is allowed for the SD, and:
+
+* A halt request may transition the hart to Debug Mode.
+* Abstract commands and program buffer execution can access state of privilege
+modes less than `M-mode`.
+* Read and Write of `Sdtrig` and `Sdext` CSRs are allowed.
+* Debugger memory accesses occur with either `S-mode` or `U-mode` privilege (as
+if `aamvirtual` = 1 and `MPP` != `M-mode`).
+
+==== `Smsdedbg` interaction with external debug security controls (Informative)
 
 [caption="Figure {counter:image}: ", reftext="Figure {image}"]
 [title= "External Debug for Supervisor Domain", id=Smsdedbg_img]
 image::images/Smsdedbg.png[]
 
-This section will be moved into the non-ISA specification for external debug
-security. It is described here as informational.
+This section will be moved into the specification for external debug security.
+It is described in this specification as informational.
 
 The `medbgen` is an enable control for external debug for the M-mode driven by
 the debug module and is expected to be established by the RoT (following RISC-V
 Security Model recommendation SR_GEN_007 and SR_GEN_012). When privilege is `M`,
 the `medbgen` gates the `haltreq` from the debug module and if is 0 prevents
```

Comment:
I presume you are specifying the additional rules in the secure external debug spec ISA - in this spec I am only specifying the msdcfg control bits and how it figures into the gating of haltreq.
I will add a reference to the ext secure debug spec (once it is available).

Reply:
It is very confusing. Because in other paragraph (above and blow), the trigger, abstract commands are mentioned. I don't think user can get the point "in this spec I am only specifying the msdcfg control bits and how it figures into the gating of haltreq."

Reply:
I can add a note to that effect in the intro such that the reader is aware that they should read the "secure external debug spec" as well to get the full picture.

Reply:
It helps little. The right direction is if the feature/behavior is mentioned in two spec, it should be described thoroughly in both.

Reply:
Again, the main spec should be the ext. secure debug spec - we cannot repeat information in 2 specs - that will lead to inconsistency and bugs - the goal of this supervisor domains spec chapter is to describe the additional requirements i.e. 2 things: 1. the controls specified in msdcfg for sdedbgalw and sdetrcalw and how they feed into/gate medbgen  2. the behavior change in m-mode to allow it to access debug CSRs when d-mode=1 so that when it is not under external debug it can manage ext debug of a SD that opts in. 

can we agree on this partitioning of the spec first? cc @joxie

Reply:
Added a note in the intro and a citation for https://github.com/riscv-non-isa/riscv-external-debug-security/blob/main/external-debug-security.pdf